lib/src/repo.rs

```diff
@@ -1312,13 +1321,31 @@ impl MutableRepo {
         Ok(rebased)
     }
 
+    /// Rebase descendants of the rewritten commits.
+    ///
+    /// The descendants of the commits registered in `self.parent_mappings` will
+    /// be recursively rebased onto the new version of their parents.
+    /// Returns a map of newly rebased commit ID to original commit ID.
+    ///
+    /// All rebased descendant commits will be preserved even if they were
+    /// emptied following the rebase operation. To customize the rebase
+    /// behavior, use
+    /// [`MutableRepo::rebase_descendants_with_options_return_map`].
     pub fn rebase_descendants_return_map(
         &mut self,
         settings: &UserSettings,
     ) -> BackendResult<HashMap<CommitId, CommitId>> {
         self.rebase_descendants_with_options_return_map(settings, Default::default())
     }
 
+    /// Rebase descendants of the rewritten commits.
+    ///
+    /// The descendants of the commits registered in `self.parent_mappings` will
+    /// be recursively rebased onto the new version of their parents.
+    /// Returns the number of rebased descendants.
+    ///
+    /// This function is similar to [`MutableRepo::rebase_descendants`], but
+    /// allows for rebase behavior to be customized via [`RebaseOptions`].
     pub fn rebase_descendants_with_options(
```

Comment:
fwiw, we have no callers of `rebase_descendants_with_options()`, and one for `rebase_descendants_return_map()` (except for tests.)

Maybe we can remove these and migrate all callers to `rebase_descendants()` and `rebase_descendants_with_options()` (`_return_map`)?

(no action needed within this PR)

Reply:
I guess those are convenience functions for the default behaviour, and perhaps they should be kept in the lib?

Reply:
My feeling is that they're too many. The simplest `rebase_descendants() -> usize` shouldn't be removed, but there aren't many callers of the other variants.